### PR TITLE
Adds notify on start toggle

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This addresses https://github.com/ansible/awx/issues/79.  It adds a toggle to the notifications list that allows the user to receive notification when their job starts.  

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
4.0.0
```

##### ADDITIONAL INFORMATION
Screen Shot 2019-06-18 at 5 14 46 PM
Screen Shot 2019-06-18 at 5 14 53 PM

### CODE REVIEW

**Comment by AlexSCorey** [comment]
recheck

File: awx/ui/client/src/notifications/shared/notification-list-init.factory.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default ['Wait', 'GetBasePath', 'ProcessErrors', 'Rest', 'GetChoices',
             scope.relatednotificationsRemove();
         }
         scope.relatednotificationsRemove = scope.$on('relatednotifications', function () {
-                var columns = ['/notification_templates_success/', '/notification_templates_error/'];
+                var columns = ['/notification_templates_started/', '/notification_templates_success/', '/notification_templates_error'];
```
Comment (jlmitch5) [inline awx/ui/client/src/notifications/shared/notification-list-init.factory.js]: looks like a trailing slash got removed by mistake from the last item in this array

File: awx/ui/client/src/notifications/shared/notification-list-init.factory.js [hunk 2]
```diff
@@ -59,12 +59,15 @@ export default ['Wait', 'GetBasePath', 'ProcessErrors', 'Rest', 'GetChoices',
...
 
-                _.map(columns, function(column){
-                    var notifier_url = url + id + column;
+            _.map(columns, function(column){
+                var notifier_url = url + id + column;
                     Rest.setUrl(notifier_url);
                     Rest.get()
                         .then(function(response) {
-                            var type = (response.config.url.indexOf('success')>0) ? "notification_templates_success" : "notification_templates_error";
+                            let checkForSuccessOrError = response.config.url.indexOf('success')>0 ? "notification_templates_success" : "notification_templates_error";
```
Comment (jlmitch5) [inline awx/ui/client/src/notifications/shared/notification-list-init.factory.js]: the `>` sign should have spaces around it.

File: awx/ui/client/src/notifications/shared/notification-list-init.factory.js [hunk 3]
```diff
@@ -59,12 +59,15 @@ export default ['Wait', 'GetBasePath', 'ProcessErrors', 'Rest', 'GetChoices',
...
+
+                            let type = response.config.url.indexOf('started')>0? "notification_templates_started" : checkForSuccessOrError
```
Comment (jlmitch5) [inline awx/ui/client/src/notifications/shared/notification-list-init.factory.js]: same here

Comment (jlmitch5) [reply awx/ui/client/src/notifications/shared/notification-list-init.factory.js]: this line is also missing a semicolon

File: awx/ui/client/src/notifications/shared/notification-list-init.factory.js [hunk 4]
```diff
@@ -59,12 +59,15 @@ export default ['Wait', 'GetBasePath', 'ProcessErrors', 'Rest', 'GetChoices',
                     callback: 'choicesReadyNotifierList'
                 });
 
-                _.map(columns, function(column){
-                    var notifier_url = url + id + column;
+            _.map(columns, function(column){
+                var notifier_url = url + id + column;
```
Comment (jlmitch5) [inline awx/ui/client/src/notifications/shared/notification-list-init.factory.js]: this line should be tabbed in 2 more

File: awx/ui/client/src/notifications/shared/toggle-notification.factory.js [hunk 5]
```diff
@@ -45,7 +45,7 @@ export default ['Wait', 'ProcessErrors', 'Rest',
                     notifier[column] = !notifier[column];
                 }
                 // Hide the working spinner
-                Wait('stop');
+            Wait('stop');
```
Comment (jlmitch5) [inline awx/ui/client/src/notifications/shared/toggle-notification.factory.js]: this line should be tabbed in 4 more

**Review by jlmitch5 — COMMENTED** [review]
looks good, some small details to change here and there.

**Review by jlmitch5 — APPROVED** [review]
looks good!

**Comment by unlikelyzero** [comment]
LGTM:

Tests executed and https://github.com/ansible/awx/issues/4113 created

**Pull request merged**